test/integration/testEasyLoggingOnConnecting.js

```diff
@@ -0,0 +1,126 @@
+/*
+ * Copyright (c) 2023 Snowflake Computing Inc. All rights reserved.
+ */
+const snowflake = require('./../../lib/snowflake');
+const connOption = require('./connectionOptions');
+const Logger = require('./../../lib/logger');
+const {reset: resetEasyLoggingModule} = require('../../lib/logger/easy_logging_starter');
+const path = require('path');
+const os = require('os');
+const fsPromises = require('fs/promises');
+const assert = require('assert');
+const logLevelBefore = Logger.getInstance().getLevel();
+const {codes} = require('./../../lib/errors');
+const errorMessages = require('./../../lib/constants/error_messages');
+let tempDir = null;
+
+describe('Easy logging tests', function () {
+
+  before(async function () {
+    tempDir = await fsPromises.mkdtemp(path.join(os.tmpdir(), 'easy_logging_connect_tests_'));
+  });
+
+  after(async function () {
+    Logger.getInstance().configure({
+      level: logLevelBefore,
+      filePath: 'snowflake.log'
+    });
+    await fsPromises.rm(tempDir, { recursive: true, force: true });
+  });
+
+  afterEach(function () {
+    resetEasyLoggingModule();
+  });
+
+  it('Should apply easy logging config when connection is being opened with callback', function (done) {
+    const logLevel = 'ERROR';
+    createConfigFile(logLevel).then((configFilePath) => {
+      const configParameters = createConfigParameters(configFilePath);
+      const connection = snowflake.createConnection(configParameters);
+      connection.connect((err) => {
+        if (err) {
+          done(err);
+        } else {
+          assert.strictEqual(Logger.getInstance().getLevelTag(), logLevel);
+          done();
+        }
+      });
+    });
+  });
+
+  it('Should fail for connecting with wrong easy logging config', function (done) {
+    const logLevel = 'something weird';
+    createConfigFile(logLevel).then((configFilePath) => {
+      const configParameters = createConfigParameters(configFilePath);
+      const connection = snowflake.createConnection(configParameters);
+      connection.connect((err) => {
+        if (err) {
+          try {
+            assert.strictEqual(err.message, errorMessages[codes.ERR_CONN_CONNECT_INVALID_CLIENT_CONFIG]);
+            assert.strictEqual(err.code, codes.ERR_CONN_CONNECT_INVALID_CLIENT_CONFIG);
```

Comment:
Code and message are here the same?

Reply:
they are different. Code is codes.ERR_CONN_CONNECT_INVALID_CLIENT_CONFIG.
Message is errorMessages[codes.ERR_CONN_CONNECT_INVALID_CLIENT_CONFIG] so we look into errorMessage array at position of given code.